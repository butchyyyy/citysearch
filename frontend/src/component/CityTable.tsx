import City from "model/City"
import React from "react"
import { Table } from "reactstrap"

interface Props {
  /** Cities to tender in the table */
  cities: City[]
}

/** Renders cities in a basic responsive table */
const CityTable = (props: Props) => (
    <Table>
      <thead>
      <tr>
        <th>City</th>
        <th>Country</th>
        <th>Population</th>
        <th>Latitude</th>
        <th>Longitude</th>
      </tr>
      </thead>
      <tbody>
      {props.cities.map((city, index) => (
          <tr key={index}>
            <td>{city.name}</td>
            <td>{city.country}</td>
            <td>{city.population}</td>
            <td>{city.lat}</td>
            <td>{city.lng}</td>
          </tr>
      ))}
      </tbody>
    </Table>
)

export default CityTable
