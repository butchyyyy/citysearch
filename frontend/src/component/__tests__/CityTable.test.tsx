import { testCities } from "__tests__/TestData"
import CityTable from "component/CityTable"
import { shallow } from "enzyme"
import React from "react"

describe("Test CityTable Component", () => {
  const wrapper = shallow(<CityTable cities={testCities} />)

  it("Test CityTable Snapshot", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
