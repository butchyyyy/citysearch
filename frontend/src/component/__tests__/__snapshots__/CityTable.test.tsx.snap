// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CityTable Component Test CityTable Snapshot 1`] = `
<Table
  responsiveTag="div"
  tag="table"
>
  <thead>
    <tr>
      <th>
        City
      </th>
      <th>
        Country
      </th>
      <th>
        Population
      </th>
      <th>
        Latitude
      </th>
      <th>
        Longitude
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="0"
    >
      <td>
        Prague
      </td>
      <td>
        Czechia
      </td>
      <td>
        1165581
      </td>
      <td>
        50.08804
      </td>
      <td>
        14.42076
      </td>
    </tr>
    <tr
      key="1"
    >
      <td>
        New Prague
      </td>
      <td>
        United States, Minnesota, Minneapolis
      </td>
      <td>
        7582
      </td>
      <td>
        44.5433
      </td>
      <td>
        -93.57607
      </td>
    </tr>
    <tr
      key="2"
    >
      <td>
        I don't have a population
      </td>
      <td>
        Some Country
      </td>
      <td />
      <td>
        0
      </td>
      <td>
        0
      </td>
    </tr>
  </tbody>
</Table>
`;
